staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -158,3 +161,184 @@ func ValidateCustomResourceDefinitionNames(names *apiextensions.CustomResourceDe
 
 	return allErrs
 }
+
+func ValidateCustomResource(customresource interface{}, crd apiextensions.CustomResourceDefinition) error {
```

Comment:
pointer.

Actually, there is no need to convert to OpenAPI in every call. Can we convert once and store the result in customResourceValidator?